### ISSUE TITLE
Chrome (Techland d3d8 game) - several issues

### ISSUE THREAD

**Issue by WinterSnowfall** [AlpyneDreams/d8vk#205]
Chrome has some issues when run with d8vk. And, ironically, other sets of issues with WineD3D. I'll list the problems it has with d8vk and discuss them separately:

- (1.) The game crashes on exit, which doesn't happen with WineD3D or with d3d8to9 + dxvk. Based on the crashlog the game generates, it looks like we're still potentially holding some extra references somewhere: [crash.log](https://github.com/AlpyneDreams/d8vk/files/14500667/crash.log)

- (2.) Trees are rendered in light blue instead of being textured properly. This also works fine both with WineD3D and d3d8to9 + dxvk.

  <details>
  <summary>WineD3D</summary>
  
  ![Trees_WineD3D](https://github.com/AlpyneDreams/d8vk/assets/6306593/96ad0d3f-e318-4b4e-9562-f9ba0f37bae1)
  
  </details>
  <details>
  <summary>d8vk</summary>

  ![Trees_d8vk](https://github.com/AlpyneDreams/d8vk/assets/6306593/b7a51343-965f-448b-b596-14093cf42757)
  
  </details>

- (3.) Ground textures are smeared ad infinitum. This works fine with WineD3D, but is also broken with d3d8to9 + dxvk.

  <details>
  <summary>WineD3D</summary>
  
  ![Ground_WineD3D](https://github.com/AlpyneDreams/d8vk/assets/6306593/b9582503-e419-46b9-87a8-7b2431321b55)
  
  </details>
  <details>
  <summary>d8vk</summary>
  
  ![Ground_d8vk](https://github.com/AlpyneDreams/d8vk/assets/6306593/6bb68a39-d272-4818-bc13-c78ef3340b92)
  
  </details>

Worthy of note is that WineD3D has problems of its own with some textures, both in the menu and in game, for example the missing crate texture in point (3.), which d8vk renders correctly.

Here's the game menu for reference:

  <details>
  <summary>WineD3D</summary>
  
  ![WineD3D](https://github.com/AlpyneDreams/d8vk/assets/6306593/2633660d-4aa4-4906-af2a-77ec488f293a)
  
  </details>
  <details>
  <summary>d8vk</summary>
  
  ![d8vk](https://github.com/AlpyneDreams/d8vk/assets/6306593/fc651d76-db3d-4924-b047-05fab3ff2383)
  
  </details>

Only point (3.) will be somewhat easy to investigate, as the game crashes when attempting to capture a d3d8 trace with apitrace. Capturing a d3d9 trace with d3d8to9 + WineD3D worked however.

In summary: (1.) and (2.) are d8vk specific, while (3.) is also a dxvk (d9vk) issue.

### Software information
Chrome 1.2.0.0 (Gold Edition) - Retail copy

### System information
- GPU: RTX 4070
- Driver: 535.161.07
- Wine version: 9.3 Staging
- DXVK version: 1a039a4

### Apitrace file(s)

This is a d3d9 trace, unfortunately: [ChromeSingle.trace.tar.xz](https://drive.google.com/file/d/1g3VSoFv5EBaONG0RvIGfM3KXKJgfykJq/view?usp=sharing)

### Log files
[ChromeSingle.log](https://github.com/AlpyneDreams/d8vk/files/14500505/ChromeSingle.log)

**Comment by WinterSnowfall** [comment]
While investigating point (1.), ironically, I stumbled into the fix for (2.) as well, and it will also fix other games that expect calls to CreateImageSurface() to succeed regardless of the format they use.

P.S.: Fun fact: the game uses `D3DFMT_R8G8B8` surfaces for lightmaps.

**Comment by WinterSnowfall** [comment]
Haven't had much luck looking into (3), so I'll note my meager findings here, lest I forget all about it.

I've looked at the sampler code in dxvk and it *seems* fine. That being said for the terrain texture in question we somehow get the first row of pixels in the mip repeated ad infinitum, rather than the full texture being tiled properly. I wasn't able to figure out why this is happening with dxvk, but simply works with WineD3D.

**Comment by WinterSnowfall** [comment]
(3) was also fixed and works with master dxvk now, most likely by @K0bin's fixed function rework.
